### PR TITLE
docs: Add `linkedProjects` section to Rust docs

### PR DESCRIPTION
**Pull request by SkymanOne**
Related to #19897

Adds a section about multi-project workspaces and how to configure rust-analyzer to diagnose them even if the cargo workspace does not list them

Release Notes:

- N/A

### CODE REVIEW

**Comment by SkymanOne** [comment]
@cla-bot check

File: docs/src/languages/rust.md [hunk 1]
```diff
@@ -163,6 +163,26 @@ Here's a snippet for Zed settings.json (the language server will restart automat
 }
 ```
 
+### Mutli-project workspaces
```
Comment (maxdeviant) [inline docs/src/languages/rust.md:166]:
```suggestion
### Multi-project workspaces
```

**Review by maxdeviant — APPROVED** [review]
Thanks!

**Comment by notpeter** [comment]
Thanks @SkymanOne!

**Pull request merged**